src/s2wsjson/validate_values_mixin.py

```diff
@@ -80,3 +48,28 @@ def to_dict(self: C) -> dict:
     def from_json(cls: Type[C], json_str: str) -> C:
         gen_model: C = cls.parse_raw(json_str)
         return gen_model
+
+
+def convert_to_s2exception(f : Callable):
+    def inner(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except (ValidationError, TypeError) as e:
+            raise S2ValidationError(args[0], 'Pydantic raised a format validation error.', pydantic_validation_error=e)
+
+    inner.__doc__ = f.__doc__
+    inner.__annotations__ = f.__annotations__
+
+    return inner
+
+
+def patch(input_class):
+
+    input_class.__init__ = convert_to_s2exception(input_class.__init__)
+    input_class.__setattr__ = convert_to_s2exception(input_class.__setattr__)
+
+    input_class.json = convert_to_s2exception(input_class.json)
```

Comment:
Can the `.json()` and `.dict()` functions produce validation errors? Any changes to the pydantic model should already have raised an exception through `init`, `setattr` or `parse_raw`. Do you see a path which can generate errors?

Reply:
You're right!
It would be required if didn't autovalidate on update (i.e. `__setattr__`).